tools/utility_functions.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+# Bash utility functions used within the project.
+
+function stage() {
+    echo
+    colored_echo BOLD_BLUE "$1"
+}
+
+function die() {
+    colored_echo RED "$*" 1>&2
+    exit 1
+}
+
+function colored_echo() {
+    local color="$1"
+    local msg="$2"
+
+    if [[ ${BASH_VERSINFO[0]} -lt 4 ]]; then
+        # Bash 3.2 distributed with macOS does not support control sequences in echo -e
+        color_code=""
+    else
+        # shellcheck disable=SC2034
+        {
```

Comment:
Why the curly brackets here?

Reply:
So that `shellcheck disable` applies to all variables.